src/i2c.c

```diff
@@ -0,0 +1,336 @@
+#include "i2c.h"
+#include "led.h"
+#include "global_vars.h"
+
+// AF bit is set when a byte transfer ends with a NACK
+static inline bool nack_recieved(void) {return I2C_SR1(I2C1) & I2C_SR1_AF;}
+static inline bool i2c_transaction_in_progress(void) {return I2C_SR2(I2C1) & I2C_SR2_BUSY;}
+
+// A timed out I2C device will NACK after receiving a byte
+#define I2C_RETURN_IF_FAILED(x) if(i2c_timed_out) { return x;}
+// The I2C protocol means that every transfer will end with either a NACK or ACK
+// The ACK requires the slave device to be actively participating in the transaction
+#define I2C_FAIL_AND_RETURN_ON_NACK(x) if(nack_recieved()) { \
```

Comment:
typo in `nack_received`

```suggestion
#define I2C_FAIL_AND_RETURN_ON_NACK(x) if(nack_received()) { \
```